From fc4b55bd7e3e2327b572b5d85f1c3d581eebd8b7 Mon Sep 17 00:00:00 2001
From: Yong Zhi <yong.zhi@intel.com>
Date: Mon, 9 Apr 2018 15:46:25 -0500
Subject: [PATCH 034/129] media: intel-ipu4: isys: Add (un)subscribe_event
 isa_sd_core_ops

Message for Open Source:
The v4l2-compliance complains about VIDIOC_(UN)SUBSCRIBE_EVENT
for "Intel IPU4 ISA" subdev, add them to fix the complaints.

Message for Internal:
[Issue/Feature]
v4l2-compliance -u /dev/v4l-subdev9

"fail: v4l2-test-controls: subscribe event for control"
"      'User Controls' failed"
"test VIDIOC_(UN)SUBSCRIBE_EVENT/DQEVENT: FAIL"

"fail: v4l2-test-controls: failed to find event for control
       'ISA enable'"
test VIDIOC_(UN)SUBSCRIBE_EVENT/DQEVENT: FAIL

[Root Cause/Changes] Missing implementation.

Change-Id: Ibbf699413009af43a7e69b88b0b888d5b6c7580a
Tracked-On: #JICCCN-87
Signed-off-by: Yong Zhi <yong.zhi@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu4/ipu4-isys-isa.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu4-isys-isa.c b/drivers/media/pci/intel/ipu4/ipu4-isys-isa.c
index a873996d..2270ee04 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-isys-isa.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-isys-isa.c
@@ -8,6 +8,7 @@
 #include <media/ipu-isys.h>
 #include <media/media-entity.h>
 #include <media/v4l2-device.h>
+#include <media/v4l2-event.h>
 #include <media/v4l2-ioctl.h>
 #include <media/videobuf2-dma-contig.h>
 
@@ -171,6 +172,8 @@ static int isa_config_vidioc_try_fmt_vid_out_mplane(struct file *file, void *fh,
 };
 
 static const struct v4l2_subdev_core_ops isa_sd_core_ops = {
+	.subscribe_event = v4l2_ctrl_subdev_subscribe_event,
+	.unsubscribe_event = v4l2_event_subdev_unsubscribe,
 };
 
 static int set_stream(struct v4l2_subdev *sd, int enable)
@@ -922,7 +925,8 @@ int ipu_isys_isa_init(struct ipu_isys_isa *isa,
 				    NR_OF_ISA_PADS,
 				    NR_OF_ISA_STREAMS,
 				    NR_OF_ISA_SOURCE_PADS,
-				    NR_OF_ISA_SINK_PADS, 0);
+				    NR_OF_ISA_SINK_PADS,
+				    V4L2_SUBDEV_FL_HAS_EVENTS);
 	if (rval)
 		goto fail;
 
-- 
1.9.1

